build_tools/benchmarks/collect_compilation_statistics.py

```diff
@@ -184,11 +184,14 @@ def main(args: argparse.Namespace):
             f"Module path isn't a module cmake target: {module_path}")
       compilation_time_ms = target_build_time_map[cmake_target]
 
-      compilation_info = CompilationInfo(model_name=benchmark_case.model_name,
-                                         model_tags=benchmark_case.model_tags,
-                                         model_source=category,
-                                         target_arch=benchmark_case.target_arch,
-                                         bench_mode=benchmark_case.bench_mode)
+      compilation_info = CompilationInfo(
+          model_name=benchmark_case.model_name,
+          model_tags=benchmark_case.model_tags,
+          model_source=category,
+          target_arch=benchmark_case.target_arch,
+          # TODO(#11071): compile_tags should be retrived from benchmark_case.compile_tags
+          # during the migration.
+          compile_tags=benchmark_case.bench_mode)
```

Comment:
So right now `BenchmarkCase` exclusively has the legacy fields. I'm confused about the relationship between `BenchmarkCase` and `BenchmarkInfo`. Is the one only for legacy benchmarks? Should its name or documentation reflect that? Can we use `compile_tags` if it exists and fall back to `bench_mode`? It seems like `bench_mode` mostly talks about runtime configuration, so it's a bit weird to include here.

Reply:
`BenchmarkInfo` is for benchmark results and `BenchmarkCase` is the internal object to store benchmark parameters. `BenchmarkInfo` is constructed from `BenchmarkCase` and some other target device spec after the benchmark run.

This change focuses on adding the new `compile_tags` in the `BenchmarkInfo` so we can record that tag in our current dashboard.

The source of "compile_tags" will be from "BenchmarkCase.run_config.module_generation_config.compile_config" once #11141 is submitted. I can wait for that change so we can populate `compile_tags` here when `BenchmarkCase.run_config` is not none.

Reply:
Refactored to use `compile_tags` from `run_config` if exists.